crates/block-producer/src/mempool/inflight_state/mod.rs

```diff
@@ -61,35 +64,19 @@ pub struct InflightState {
     chain_tip: BlockNumber,
 }
 
-/// The aggregated impact of a set of sequential transactions on the [InflightState].
-#[derive(Clone, Default, Debug, PartialEq)]
-struct StateDelta {
-    /// The number of transactions that affected each account.
-    account_transactions: BTreeMap<AccountId, usize>,
-
-    /// The nullifiers consumed by the transactions.
+#[derive(Clone, Debug, PartialEq)]
+struct Delta {
+    account: AccountId,
     nullifiers: BTreeSet<Nullifier>,
-
-    /// The notes produced by the transactions.
     output_notes: BTreeSet<NoteId>,
 }
 
-impl StateDelta {
-    fn new(txs: &[AuthenticatedTransaction]) -> Self {
-        let mut account_transactions = BTreeMap::<AccountId, usize>::new();
-        let mut nullifiers = BTreeSet::new();
-        let mut output_notes = BTreeSet::new();
-
-        for tx in txs {
-            *account_transactions.entry(tx.account_id()).or_default() += 1;
-            nullifiers.extend(tx.nullifiers());
-            output_notes.extend(tx.output_notes());
-        }
-
+impl Delta {
+    fn new(tx: &AuthenticatedTransaction) -> Self {
```

Comment:
nit: maybe this should be `From<&AuthenticatedTransaction> for Delta`?

Reply:
I'm not a fan of `From/Into` for references tbh, they're pretty hard to discover naturally.